mod/photos.php

```diff
@@ -282,77 +258,61 @@ function photos_post(App $a)
 
-			// find and delete the corresponding item with all the comments and likes/dislikes
-			Item::deleteForUser(['resource-id' => $res, 'uid' => $page_owner_uid], $page_owner_uid);
+				// find and delete the corresponding item with all the comments and likes/dislikes
+				Item::deleteForUser(['resource-id' => $res, 'uid' => $page_owner_uid], $page_owner_uid);
 
-			// Update the photo albums cache
-			Photo::clearAlbumCache($page_owner_uid);
+				// Update the photo albums cache
+				Photo::clearAlbumCache($page_owner_uid);
+				notice(L10n::t('Album successfully deleted'));
+			} else {
+				notice(L10n::t('Album was empty.'));
+			}
 		}
 
-		$a->internalRedirect('photos/' . $a->data['user']['nickname']);
-		return; // NOTREACHED
-	}
-
-
-	// Check if the user has responded to a delete confirmation query for a single photo
-	if ($a->argc > 2 && !empty($_REQUEST['canceled'])) {
-		$a->internalRedirect($_SESSION['photo_return']);
+		$a->internalRedirect('photos/' . $a->argv[1]);
 	}
 
-	if ($a->argc > 2 && defaults($_POST, 'delete', '') === L10n::t('Delete Photo')) {
-
-		// same as above but remove single photo
-
-		// Check if we should do HTML-based delete confirmation
-		if (!empty($_REQUEST['confirm'])) {
-			$drop_url = $a->query_string;
-
-			$a->page['content'] = Renderer::replaceMacros(Renderer::getMarkupTemplate('confirm.tpl'), [
-				'$method' => 'post',
-				'$message' => L10n::t('Do you really want to delete this photo?'),
-				'$extra_inputs' => [],
-				'$confirm' => L10n::t('Delete Photo'),
-				'$confirm_url' => $drop_url,
-				'$confirm_name' => 'delete', // Needed so that confirmation will bring us back into this if statement
-				'$cancel' => L10n::t('Cancel'),
-			]);
-
-			$a->error = 1; // Set $a->error so the other module functions don't execute
-			return;
+	if ($a->argc > 3 && $a->argv[2] === 'image') {
+		// Check if the user has responded to a delete confirmation query for a single photo
+		if (!empty($_POST['canceled'])) {
+			$a->internalRedirect('photos/' . $a->argv[1] . '/image/' . $a->argv[3]);
 		}
 
-		if ($visitor) {
-			$r = q("SELECT `id`, `resource-id` FROM `photo` WHERE `contact-id` = %d AND `uid` = %d AND `resource-id` = '%s' LIMIT 1",
-				intval($visitor),
-				intval($page_owner_uid),
-				DBA::escape($a->argv[2])
-			);
-		} else {
-			$r = q("SELECT `id`, `resource-id` FROM `photo` WHERE `uid` = %d AND `resource-id` = '%s' LIMIT 1",
-				intval(local_user()),
-				DBA::escape($a->argv[2])
-			);
-		}
+		if (!empty($_POST['delete'])) {
+			// same as above but remove single photo
+			if ($visitor) {
+				$r = q("SELECT `id`, `resource-id` FROM `photo` WHERE `contact-id` = %d AND `uid` = %d AND `resource-id` = '%s' LIMIT 1",
+					intval($visitor),
+					intval($page_owner_uid),
+					DBA::escape($a->argv[3])
+				);
+			} else {
+				$r = q("SELECT `id`, `resource-id` FROM `photo` WHERE `uid` = %d AND `resource-id` = '%s' LIMIT 1",
```

Comment:
Is a easy replacement with `DBA::p` possible?

Reply:
Same as above.